internal/experiments/chaos_controller/chaosmesh.go

```diff
@@ -0,0 +1,232 @@
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+type ChaosController struct {
+	cluster_config_path string
+	pod                 *pods.Pods
+	dynamicClient       *dynamic.DynamicClient
+	namespace           string
+	curExp              *unstructured.Unstructured
+	gvr                 *schema.GroupVersionResource
+}
+
+// New creates a new podkiller
+func New(namespace string) *ChaosController {
+	c, err := client.New()
+	p := pods.New(c)
+	if err != nil {
+		return nil
+	}
+
+	config_path := getConfigPath()
+
+	config, err := clientcmd.BuildConfigFromFlags("", config_path)
+	if err != nil {
+		log.Fatalf("Failed to build kubeconfig: %v", err)
+	}
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("Failed to create Kubernetes client: %v", err)
+	}
+
+	return &ChaosController{config_path, p, dynamicClient, namespace, nil, nil}
+}
+
+// configPath    - path to k8s cluster config
+// killNodes 	- to kill or not to kill affected nodes after each run
+// recoveryTime - how much time to wait for cluster to recover after experiment
+func (cc *ChaosController) RunChaosExperiment(configPath string, restartNodes bool, recoveryTime string) error {
+	// Work with experiment yaml file
+	experiment_file, err := os.ReadFile(configPath)
+	if err != nil {
+		log.Fatalf("Failed to read manifest file: %v", err)
+	}
+	reader := bytes.NewReader(experiment_file)
+	decoder := yaml.NewYAMLOrJSONDecoder(reader, 1000)
+	parsedYaml := &unstructured.Unstructured{}
+
+	if err := decoder.Decode(parsedYaml); err != nil {
+		log.Fatalf("Failed to decode YAML: %v", err)
+	}
+	cc.curExp = parsedYaml
+
+	exp_type, _, err := unstructured.NestedString(cc.curExp.Object, "kind")
+	if err != nil {
+		return err
+	}
+	gvr := schema.GroupVersionResource{
+		Group:    "chaos-mesh.org",          // ChaosMesh group
+		Version:  "v1alpha1",                // Version of the ChaosMesh resource
+		Resource: strings.ToLower(exp_type), // Resource is "podchaos"
+	}
+	cc.gvr = &gvr
+
+	// Apply the resource (equivalent to `kubectl apply -f`)
+	_, err = cc.dynamicClient.Resource(gvr).Namespace(cc.namespace).Create(context.TODO(), parsedYaml, v1.CreateOptions{})
+	if err != nil {
+		log.Fatalf("Failed to apply ChaosMesh resource: %v", err)
+	}
+
+	cc.ExperimentSleep(recoveryTime)
+
+	if restartNodes {
+		RestartAffectedPods(cc)
+	}
+	return err
+}
+
+// Deleting the experiment from chaosmesh list
+// to avoid collisions, when the same experiment is started
+func (cc *ChaosController) ClearExperimentData(chaosType string) {
+	gvr := schema.GroupVersionResource{
+		Group:    "chaos-mesh.org",           // ChaosMesh group
+		Version:  "v1alpha1",                 // Version of the ChaosMesh resource
+		Resource: strings.ToLower(chaosType), // Resource is "podchaos"
+	}
+
+	err := cc.dynamicClient.Resource(gvr).Namespace(cc.namespace).DeleteCollection(
+		context.TODO(),
+		v1.DeleteOptions{},
+		v1.ListOptions{},
+	)
```

Comment:
Field `ChaosController.gvr` is write-only for now, and it already stores resource name.  

Either remove `cc.gvr = gvr` from `RunChaosExperiment`
Or reuse stored gvr (and run this function before starting next experiment):
```suggestion
func (cc *ChaosController) ClearLastExperimentData() {
        if cc.gvr == nil { return }
	err := cc.dynamicClient.Resource(cc.gvr).Namespace(cc.namespace).DeleteCollection(
		context.TODO(),
		v1.DeleteOptions{},
		v1.ListOptions{},
	)
        cc.gvr = nil
```